### PR TITLE
Service Borg Upgrade

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds an upgrade module for service borgs to the exofab, printable with Biotech 2, Engineering 3, and Data Theory 2.
The upgrade gives service borgs a large beaker, a kitchen knife, and a borg food bag capable of emptying into microwaves.

Service borgs can now use microwaves, though all other silicon models remain unable to.

Service borgs can now make use of brewers, soda dispensers, and booze dispensers, though they still cannot use chemistry dispensers.

### CODE REVIEW

**Comment by Unfit** [comment]
:+1: :+1: :+1: :+1: :+1:
Amazing!

**Comment by IratePirate** [comment]
It's always needed some love. Good man.

**Comment by Probe1** [comment]
I don't see any balance concerns for rogue/malf and think it's a solid change.

File: code/game/machinery/kitchen/microwave.dm [hunk 1]
```diff
@@ -143,16 +143,16 @@
...
-		for (var/obj/item/weapon/reagent_containers/food/snacks/grown/G in O.contents)
+		for (var/obj/item/weapon/reagent_containers/food/snacks/G in O.contents)
 			B.remove_from_storage(G,src)
 			if(contents && contents.len >= limit) //Sanity checking so the microwave doesn't overfill
 				user << "You fill the Microwave to the brim."
 				break
 
 		if(!O.contents.len)
-			user << "You empty the plant bag into the Microwave."
+			user << "You empty the [O] into the Microwave."
```
Comment (ComicIronic) [inline code/game/machinery/kitchen/microwave.dm]: \the [O]

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 2]
```diff
@@ -288,11 +288,42 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 	ui_interact(user)
 //Cafe stuff
 
-/obj/machinery/chem_dispenser/brewer/
+/obj/machinery/chem_dispenser/nonmedical/
```
Comment (ComicIronic) [inline code/modules/reagents/Chemistry-Machinery.dm]: Paths are actually the bad way of doing this.
You should be making a can_use(robot) proc and using that instead in the base attackby() code for dispensers.
That's OOP.

File: code/game/machinery/kitchen/microwave.dm (hunk 1 above)
Comment (Shadowmech88) [inline code/game/machinery/kitchen/microwave.dm]: Fixed. What's the difference between the and \the?

Comment (ComicIronic) [inline code/game/machinery/kitchen/microwave.dm]: \the formats for proper and improper nouns. So \the [x] is Rob, not the Rob, but the toolbox, not toolbox. Just the doesn't format.

**Comment by Shadowmech88** [comment]
I think I somehow re-committed every commit made since I made the PR.
Oops.

**Comment by Shadowmech88** [comment]
I'm told I should just make a new PR instead of trying to fix this one.

**Comment by Probe1** [comment]
panic
squash
fuck up
picnic

**Pull request closed without merging**